docs/proposals/versioning-and-release.md

```diff
@@ -0,0 +1,72 @@
+# Devfile Versioning and Release Process
+This design document outlines the proposed versioning and release process for the Devfile spec. 
+
+This document summarizes parts from he Devfile API technical meeting slides presented by @davidfestal back in October, and I recommend having a read through of it: https://docs.google.com/presentation/d/1ohM1HzPB59a3ajvB7rVWJkKONLBAuT0mb5P20_A6vLs/edit#slide=id.g8fc722bef9_1_8
+
+## Versioning
+
+The following sections outline how we version the Devfile Kubernetes API, as well as the JSON schema generated from the API.
+
+### Kubernetes API
+The Devfile Kubernetes API (defined in https://github.com/devfile/api/) is the single source of truth for the Devfile spec. It’s defined under `pkg/apis/` and contains the Go structs that are used by the devworkspace operator and devfile library. It's also where the Devfile JSON schema is generated from. 
+
+**Versioning Scheme**: [Kubernetes](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning) (e.g. v1alpha1, v1beta2, v1, v2, etc)
+
+**Update Process**: 
+
+   1) Add a new folder in the source repository (`pkg/apis/workspaces` in `devfile/api`)
+   2) Add a schema and version in the CRD manifests
+   3) Generate the JSON schema from the API. New JSON schema will be located under `schemas/latest`.
+   4) Update the devworkspace operator and devfile library to consume the Go structs in the new API version, as needed.
+
+**When to Update?**  As incrementing the Kubernetes API version for Devfile is a relatively heavy process, and affects the library, only update the K8s API version when absolutelyneeded 
+   - for **big** changes or backwards incompatible changes.
+
+### Devfile JSON Schema
+
+As mentioned above, the Devfile JSON schema is generated from the Go structs defined in the Devfile Kubernetes API. The latest JSON schema for a given API version is located under `schemas/<api-version>` in the [devfile/api repo](https://github.com/devfile/api/). 
+
+**Versioning Scheme**: Semver (major.minor.bugfix)
+
+**Update Process**: 
+  
+   1) Update the schema version string in the `// +devfile:jsonschema:version=<schema-version>` annotation in `pkg/apis/workspace/<k8s-api-version>/doc.go`
+   2) Re-generate the json schema
+   3) Publish new schema to devfile.io website
+
+**When to Update?** On each release of the schema, incremented based on the changes going in to the release (e.g. bugfix = comments / bugfixes, minor = larger and / or backwards compatible changes, major == breaking / backwards incompatible changes). K8S API version updates should also result in an appropriate increment of the schema version.
```

Comment:
Consider changing to a list for easy reading. The major entry is using == instead of =.

Reply:
Sounds good to me, updated.